presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -645,6 +646,7 @@ private static void addLookupOuterDrivers(LocalExecutionPlanContext context)
         private int nextOperatorId;
         private boolean inputDriver = true;
         private OptionalInt driverInstanceCount = OptionalInt.empty();
```

Comment:
Have you considered to reuse `driverInstanceCount`, it is being set when needed and does not carry over by `LocalExecutionPlanContex::createSubContext`

Reply:
That was my initial approach, but it turned out to be quite fragile because it can only be set once per context.

Reply:
Do you mean `setDriverInstanceCount` can only be called once?